docs/proposals/image-promotion.md

```diff
@@ -0,0 +1,152 @@
+
+As a developer of a Rails application, when I update the source code and OpenShift
+build my image I want to promote the image by passing it throught QA and devops team.
+
+#### Example
+
+* The image is passed to a remote QA team by pushing the image to a shared Docker registry
+* Remote QA team pull the image and drive the testing and verification
+* Remote QA team push the verified image back to the shared Docker registry, tagged as "verified" and with additional sign-off LABEL(s).
+* Remote devops team pull the image from shared Docker registry and make it available in "stage" cluster.
+* Remote devops team promotes the image from "stage" to "production"
+
+### Manual promotion of builder image using external Docker registry
+
+As a platform operator, I want to be able to promote the S2I builder image(s) when a new
+version is available in Red Hat Docker registry. I want to perform additional verifications
+(iow. deploy to "stage" first) before rolling a new image out to production which causes rebuild
+of thousand applications.
+
+#### Example
+
+* OpenShift engineering push the `openshift/ruby-22-rhel7` image to internal CI registry.
+* OpenShift QA team pull the image from the internal CI registry and run tests.
+* OpenShift QA team sign-off the Docker image when tests pass and push it to "integration" registry.
+* OpenShift operations team pulls the image from "integration" registry and promote it to "stage" environment.
+* OpenShift operations team manually promote the image to production.
+
+### Semi-automatic promotion and testing of builder images
+
+As a OpenShift S2I image developer, I want to have the image rebuilt, tested and pushed to
+the Docker Hub or to internal registry and tag it as "qa-ready" and notify the QA team about
+availability of a new Docker Image.
+
+#### Example of pull request tester
+
+* OpenShift engineering propose a change to `openshift/ruby-22-rhel7` image.
+* CI starts OpenShift server and execute the extended tests cases related to this image.
+* CI send notification to original pull request.
+
+#### Example of push_images job
+
+* OpenShift engineering merge change to `openshift/ruby-22-rhel7` repository.
+* OpenShift server detect the change and execute test suite for given image.
+* If the image is 'base' image, OpenShift server launch multiple tests for all dependent images.
+* OpenShift server push the image to internal registry and Docker Hub (see: "Manual promotion of builder image")
+
+## Areas of improvement
+
+The first use-case describes a typical Continuos Delivery example, where the developer uses
+both the Jenkins server and the OpenShift to deliver updates to the application.
+
+### Jenkins
+
+* There is no support for enabling the Jenkins when creating a new application.
+* There is no official "jenkins" plugin for OpenShift available.
+* Users have to manually tweak the Jenkins job parameters.
+* The Jenkins job definition is written in shell and it is complex.
```

Comment:
This does not cover the scenario if you have 1) external jenkins or 2) you run jenkins on different cluster. In that case it might be not trivial to export/import the secrets to the service account.

I think what Ben is suggesting here is a Jenkins plugin that will ask you for OpenShift credentials  (either token or certificate or kubeconfig) as part of the Jenkins job definition and make it available as environment variable for the job and for the `oc` command.

However, this can be solved by parametrized build, but I'm not sure if you willing to provide certificates or tokens there.